### PR TITLE
feat(gwas catalog sumstats): finemapping

### PR DESCRIPTION
**Pull request by project-defiant**
# Context

We want to perform locus breaker clumping and SuSiE finemapping on harmonised summary statistics comming from GWAS Catalog.

## Implementations

This PR implements:
- [x] Dag for performing `locus_breaker_clumping` and `study_index` generation for GWAS Catalog summary statistics.
- [x] Dag for finemapping clumped results.
- [x] Next iteration of documentation for the GWAS Catalog.
- [x] Refactored finemapping operator to take into accout already finemapped loci and limit for the batch jobs to call

>[!NOTE]
> **Locus Breaker Clumping performance**
> The performance of LB clumping was not ideal. The step took ~2h to compute the StudyLocus starting from 69K harmonised summary statistics. [See dataproc job](https://console.cloud.google.com/dataproc/jobs/4a6990a9-b480-4a27-a244-a57e22100c88/monitoring?region=europe-west1&project=open-targets-genetics-dev).
> This situation is a partially the result of the largely distributed dataset - see the first spike in nodes representing the first job to list all parquet files in subdirectories.
>
> image
> The number of loci resulted from clumping oscilated ~440K. 

Running code with this branch we were able to perform the fine-mapping of the 441k loci in  7h.

The way how the finemapping works:

1. list all loci outputed from locus breaker
2. list all log files from previous finemapping runs
3. make a diff and submit the jobs

This approach is not ideal due to the number of google API calls (knowledge post mortem - see distrubution of the calls in the buckets on [23rd of October](https://console.cloud.google.com/monitoring/dashboards/builder/f580f0d6-15fb-430a-b313-88ca8266525d;startTime=2024-10-22T22:57:27.055Z;endTime=2024-10-23T22:54:00.850Z?project=open-targets-genetics-dev&pageState=(%22eventTypes%22:(%22selected%22:%5B%22CLOUD_ALERTING_ALERT%22%5D)))  ) we need to make when running `list.objects`, the better solution would be to:

1. generate the manifests
2. submit the batch jobs in consecutive order depending on the manifest
3. cache the information if the manifest was used by finemapping job or not

This could be implemented as an enhancement in the future.

### CODE REVIEW

**Comment by project-defiant** [comment]
@addramir ignore the docs for now. I am changing them as soon as the dag is successful

File: src/ot_orchestration/dags/config/gwas_catalog_sumstats_susie_clumping.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+dataproc:
+  python_main_module: gs://genetics_etl_python_playground/initialisation/gentropy/dsdo_top_hits_step/cli.py
```
Comment (DSuveges) [inline src/ot_orchestration/dags/config/gwas_catalog_sumstats_susie_clumping.yaml]: Very minor, but I would change all branch specific reference to `dev`.

Comment (project-defiant) [reply src/ot_orchestration/dags/config/gwas_catalog_sumstats_susie_clumping.yaml]: you are right, thanks!

**Review by DSuveges — APPROVED** [review]
Thanks for addressing my comment. All looks good and sensible, however I have to admit, I haven't run them. :D

**Comment by project-defiant** [comment]
@DSuveges thank you!

**Pull request merged**